### PR TITLE
docs: only use type annotations for documentation

### PR DESCRIPTION
**Pull request by nejch**
We probably still have a lot of incorrect types in the docstrings and this should just take care of it in the future.

| Before | After |
|--------|------|
| image | image |

Clashes with the internal extension though so needs more work

### CODE REVIEW

**Comment by JohnVillalovos** [comment]
Looks cool! Thanks for this. I'll review once you mark it ready for review.

**Comment by nejch** [comment]
Turns out I was just using an outdated external module but it's now part of `autodoc` so it was just a matter of adding a flag. This looks a bit more like the original now, but pulls typehints from code. Should be ready now.

File: docs/conf.py [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "sphinx.ext.autodoc",
     "sphinx.ext.autosummary",
     "ext.docstrings",
+    "sphinx_autodoc_typehints",  # must be loaded after napoleon modules
```
Comment (JohnVillalovos) [inline docs/conf.py]: What is a "napoleon" module? Is it one of the ones above?

File: gitlab/client.py [hunk 2]
```diff
@@ -269,7 +269,7 @@ def version(self) -> Tuple[str, str]:
         object.
 
         Returns:
-            tuple (str, str): The server version and server revision.
+            tuple: The server version and server revision.
                               ('unknown', 'unknwown') if the server doesn't
```
Comment (JohnVillalovos) [inline gitlab/client.py:273]: While in here, maybe do:  s/unknwown/unknown/  to fix the mis-spelling.

**Review by JohnVillalovos — APPROVED** [review]
I'm okay to merge it as is.

Thanks for doing this.

File: docs/conf.py (hunk 1 above)
Comment (nejch) [reply docs/conf.py]: yes, `ext.docstrings` was extending google's modules but this is no longer needed now as I didn't need to add the extra module in the end.

File: gitlab/client.py (hunk 2 above)
Comment (nejch) [reply gitlab/client.py:273]: Oh, thanks. Even the `tuple` here is not needed 🤔  seems like my search/replace missed a few of these, I'll check again.

Comment (nejch) [reply gitlab/client.py]: https://github.com/python-gitlab/python-gitlab/pull/1709/commits/79e785e765f4219fe6001ef7044235b82c5e7754 also cleans up return types and a bit of formatting.

**Review by JohnVillalovos — APPROVED** [review]
Looks good to me.

Please merge whenever you want to.

**Pull request merged**